### PR TITLE
Feature/support sites

### PR DESCRIPTION
**Pull request by kurtmc**
So I have been playing around with adding the `site_name` attribute to some of the resources and data sources. I just wanted to open up a PR early as I want to know what the right direction for this change is. At the moment it's a bit messy, I hope you can point me in the right direction.

I have made it so that if you do not specify the `site_name` attribute it defaults to the old behaviour of choosing the site name set on the provider.

### CODE REVIEW

**Comment by paultyng** [comment]
In general this is what I would expect, a few notes so far:

- We should use the same name in these as we do on the provider schema, its called `site` there, we could change that to `site_name` also, but a breaking change
- You'll want to set these as `Computed` and set them in the methods that set all the attributes to the value used in the API request (so may need a new param on those funcs or something).

File: controller.sh [hunk 1]
```diff
@@ -52,7 +52,7 @@ case "$1" in
   "reset")
     git checkout - testdata/unifi/
     for file in $( git ls-files --others --exclude-standard | grep testdata/unifi ) ; do
-            rm -f ${file}
+            rm -f ${file} || sudo rm -f ${file}
```
Comment (kurtmc) [inline controller.sh:55]: I am not that fussed about this change, happy to remove it from this PR if you don't want it.

Comment (paultyng) [inline controller.sh:55]: Yeah lets just leave this, can always run the script with sudo I think?

```suggestion
            rm -f ${file}
```

**Review by paultyng — APPROVED** [review]
LGTM, I'd like more testing on all the resources to cover non-default site, but we can add that in follow-ups or as we experience weirdness.

**Comment by paultyng** [comment]
I'm going to cut a beta release with this to test out, can promote it once we are sure it looks good.

**Pull request merged**